### ISSUE TITLE
Create a new GET /courses_user endpoint

### ISSUE THREAD

**Issue by kyleecodes** [chaynHQ/bloom-backend#720]
### Overview
To optimize user data loading, introduce a new `GET /courses_user` endpoint that retrieves session and course data specifically for courses or session pages. This means we can remove courses and user data from the profile. 

**This issue is not blocked, but ideally #717 should be completed first.**

### Action Items
- [ ] Create a new `GET /courses_user` endpoint in the backend.
- [ ] Implement the `getCourseUserByUserId` service method for this endpoint.
- [ ] Write backend tests to ensure proper functionality (alternatively, this can be made into a separate issue.)
- [ ] Add a new controller file for handling this endpoint.
- [ ] Include code comments.

**Comment by csehatt741** [comment]
@kyleecodes, I`m also happy to help here.
